client.go

```diff
@@ -33,24 +33,33 @@ var (
 	errorAddingServerCertificateToPool = errors.New("Error adding trusted server certificate to pool.")
 )
 
+// Mender API Client wrapper. A standard http.Client is compatible with this
+// interface and can be used without further configuration where ApiClient is
+// expected. Instead of instantiating the client by yourself, one can also use a
+// wrapper call NewApiClient() that sets up TLS handling according to passed
+// configuration.
+type ApiClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
 type RequestProcessingFunc func(response *http.Response) (interface{}, error)
 
 // Client initialization
-func NewHttpClient(conf httpsClientConfig) (*http.Client, error) {
+func NewApiClient(conf httpsClientConfig) (*http.Client, error) {
 
 	if conf == (httpsClientConfig{}) {
-		return newHttpClient(), nil
+		return newApiHttpClient(), nil
 	}
 
-	return newHttpsClient(conf)
+	return newApiHttpsClient(conf)
 }
 
-func newHttpClient() *http.Client {
+func newApiHttpClient() *http.Client {
```

Comment:
I think the old name was better here. In fact we are returning JUST http client.